main.go

```diff
@@ -32,14 +32,19 @@ func main() {
 		insecureListenAddress  string
 		upstream               string
 		label                  string
+		headerName             string
```

Comment:
There's a mix of `headerName` and `header` in this file. `headerName` is filled with the flag, but `header` is validated and actually used which - obviously - is always empty.

Reply:
You are right I had to use the following patch

```
diff --git a/main.go b/main.go
index d7b29dc..3628eaa 100644
--- a/main.go
+++ b/main.go
@@ -32,7 +32,6 @@ func main() {
                insecureListenAddress  string
                upstream               string
                label                  string
-               headerName             string
                queryParam             string
                enableLabelAPIs        bool
                unsafePassthroughPaths string // Comma-delimited stri
@@ -44,7 +43,7 @@ func main() {
        flagset.StringVar(&insecureListenAddress, "insecure-listen-ad
        flagset.StringVar(&upstream, "upstream", "", "The upstream UR
        flagset.StringVar(&queryParam, "query-param", "", "The query 
-       flagset.StringVar(&headerName, "header", "", "The HTTP header
+       flagset.StringVar(&header, "header", "", "The HTTP header nam
        flagset.StringVar(&label, "label", "", "The label to enforce 
                "This label will be also required as the URL paramete
                " make it required for this proxy to have URL in form
```